### DISCUSSION TITLE
How does the database design work?

### DISCUSSION THREAD

**Discussion by NeilVanGotham** [GrandComicsDatabase/gcd-django#603] [Q&A] [answered]
I don't know if this is the right place to ask this question.
I downloaded the sql dump from [https://www.comics.org/download/](https://www.comics.org/download/) to use in my own project, compliant with the data distribution policy of course. But how is the database designed? I don't see a characters table, but I do see a characters content type.

**Comment by adia** [comment, marked as answer]
Hi Neil! You might want to start from the info linked at the following URL:

https://docs.comics.org/wiki/Main_Page#Database

As mentioned there, the Django models (found in apps/gcd/models/*.py in our source code) are the ground truth for the database schema - we don't manage the SQL directly. We don't maintain an up-to-date db-level description of all the fields and their relationships, but the descriptions found at https://docs.comics.org/wiki/Formatting_Documentation will help you. Feel free to ask about any details you'd like to clarify.

About the characters table in particular, while we are in a process of migration from text-based character records to dedicated db tables linked by foreign keys to story records, the database schema hasn't been finalized yet in all details so we have been omitting these tables from our public dumps. The latest pending change is adding a universe table for characters that require this information, and I believe after that we should be able to add the character tables to the dump.

**Reply by NeilVanGotham** [reply]
Thank you, Adia! I will be looking for an alternative for now

**Comment by jochengcd** [comment]
the characters objects are now in the public dumps
